src/etc/inc/plugins.inc.d/core.inc

```diff
@@ -82,16 +82,48 @@ function core_services()
         $services[] = $pconfig;
     }
 
-    if (is_apinger_enabled()) {
-        $pconfig = array();
-        $pconfig['name'] = "apinger";
-        $pconfig['description'] = gettext("Gateway Monitoring Daemon");
-        $pconfig['php']['restart'] = array('setup_gateways_monitor');
-        $pconfig['php']['start'] = array('setup_gateways_monitor');
-        $pconfig['pidfile'] = '/var/run/apinger.pid';
-        $services[] = $pconfig;
-    }
+    if (!isset($config['system']['prefer_dpinger'])) {
```

Comment:
if you push this to is_apinger_enabled and is_dpinger_enabled this doesn't have to be evaluated here and dpinger plugs in seamlessly here

Reply:
just belts and braces... but I'll remove it if you think it's unnecessary.

Reply:
I've moved the catches over to the services...